### ISSUE TITLE
[feature]: Category Sorting with Venia (PWA)

### ISSUE THREAD

**Issue by ericerway** [magento/pwa-studio#360]
## This issue is for the following packages:

[X] `venia-concept`
[ ] `pwa-buildpack`
[ ] `peregrine`
[ ] `pwa-module`
[ ] `pwa-devdocs`

## Description

https://magento.invisionapp.com/share/WQN5F7BYBPG#/324119989_Product_Sort

### JIRA issue for design
https://jira.corp.magento.com/browse/UXD-181

Screen: Category with Filter and Sort
![image](https://user-images.githubusercontent.com/2686596/46971332-4ada9e00-d081-11e8-9d24-86e978ad91a3.png)

Screen: Sort By with Category
![image](https://user-images.githubusercontent.com/2686596/46971294-372f3780-d081-11e8-8384-547500a34a1d.png)

As a shopper, I want to filter and sort products within a category so that I can make a purchase on Venia.

- Price: Low to High
- Price: High to Low
- Featured
- Newest
- Best Selling
- Ratings: High to Low (non-MVP)

#### Assumptions
- Dependent on Category page and ability to retrieve products by category
- Ratings not part of MVP (concept only)
- Sorting will be done SSR
- Dependency on collection filtering variable support; follow-up with graphql-ce
 
## Tasks
- [ ] Discovery for GraphQL coverage w/r/t searching and filtering queries (input types)
- [ ] Discovery for how sorting affects and shown with URL (linking+routing)
- [ ] Implement Sort By panel with basic options (not Ratings)
- [ ] Testing and documentation (https://github.com/magento-research/pwa-studio/wiki/Documentation-Guidelines)

**Comment by ericerway** [comment]
Kamino cloned this issue to [magento-research/pwa-studio](https://github.com/magento-research/pwa-studio/issues/361)

**Comment by JStein92** [comment]
UI is done on this - Currently waiting for https://github.com/magento/graphql-ce/pull/259 to be fixed/merged so we can use query variables for sorting

**Comment by cherdman** [comment]
Blocked by https://github.com/magento/graphql-ce/pull/259

**Comment by awilcoxa** [comment]
GraphQL dependency is marked as merged, appears that progress on this issue should resume. Moving back to ready to start for evaluation.

**Comment by JStein92** [comment]
@awilcoxa I believe this is still not possible. Variable queries using Objects is in 2.3.1, not 2.3.0.

**Comment by awilcoxa** [comment]
@JStein92 good call out, putting issue back on hold.

**Comment by supernova-at** [comment]
@JStein92 you mentioned

> UI is done on this

Is there an open PR that should be attached to this issue?

**Comment by awilcoxa** [comment]
Revisit of previously completed design and UI from previous PR, needs evaluation of GraphQL  dependency

**Comment by cherdman** [comment]
https://jira.corp.magento.com/browse/PWA-73

**Comment by soumya-ashok** [comment]
Scott and I have reviewed sort, some design changes need to be made. Will be working on that this week @awilcoxa .

**Comment by larsroettig** [comment]
@awilcoxa i will work on it

**Comment by soumya-ashok** [comment]
@larsroettig Thanks for picking this up. @schensley will be sharing the desktop mocks and working with you on implementation.

Mobile XD prototype : https://xd.adobe.com/view/15934509-f803-4dc8-7075-885b7e9e7407-63a2/

Screens:
![Product Sort](https://user-images.githubusercontent.com/25960021/73561306-5f7da980-441e-11ea-8be1-37c6ef5929c4.png)

![Product Sort – 2](https://user-images.githubusercontent.com/25960021/73561313-63113080-441e-11ea-8370-59495cf88a14.png)

![Product Sort – 3](https://user-images.githubusercontent.com/25960021/73561324-673d4e00-441e-11ea-8b95-eb0919084edb.png)

![Product Sort – 4](https://user-images.githubusercontent.com/25960021/73561339-702e1f80-441e-11ea-8a79-061033e07e4a.png)

**Comment by schensley** [comment]
Sorting for desktop
![Venia_Cart_Desktop_00](https://user-images.githubusercontent.com/19538346/73568184-898a9800-442d-11ea-801d-4431a6bad6ed.png)
![Venia_Cart_Desktop_01_SortingOptions](https://user-images.githubusercontent.com/19538346/73568186-898a9800-442d-11ea-83ae-82403c982523.png)
![Venia_Cart_Desktop_02_HoverState](https://user-images.githubusercontent.com/19538346/73568187-898a9800-442d-11ea-9264-3deabd6fe928.png)

**Comment by gamul** [comment]
The same button "Filter" opens an MODAL section. Early mocks for sorting were also with modal bottom drawer. Why have you changed your mind?!
Why do we get new UI element instead of stylish, unified and predictable UI?!
